core/src/test/resources/sql/sub-query.iq

```diff
@@ -3685,4 +3685,15 @@ FROM dept d1;
 
 !ok
 
+# [CALCITE-6041] MAP sub-query gives NullPointerException
+SELECT map(SELECT empno, deptno from emp where deptno < 20);
```

Comment:
test may be nondeterministic. if Calcite allows an `order by` in the subquery I think you should add it

also add tests where the map contains 1 row and 0 rows

Reply:
Leave it in. I agree that it will be deterministic, knowing where the test data comes from.

I think we need tests with N, 1 and 0 rows.

Reply:
ok, I have rebased the main and squashed the commits. If you have time, PTAL~ thanks.